icefoganalytics/travel-authorization
# Fix tsconfig.json Error Around `express-unless`

Fixes https://github.com/icefoganalytics/travel-authorization/issues/151

Relates to:

- https://github.com/auth0/express-jwt/issues/307

# Context

**Describe the bug**
Go to this file `api/tsconfig.json` and see that it is showing as errored. This error doesn't seem to have any effect, but it has the potential to hide meaningful errors and creates noise when developing.

```
Cannot find type definition file for 'express-unless'.
  The file is in the program because:
    Entry point for implicit type library 'express-unless'
```

Solution is likely to upgrade express-jwt. See https://github.com/auth0/express-jwt/issues/307

**Screenshots**
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/f4c66332-8e97-40b5-96d6-55126e44764b)

# Implementation

Upgrade express-jwt to fix tsconfig.json error.
Upgrade jwks-rsa to match example from documentation.
Upgrade uuid to fix warning.
Use type casting to make jwksRsa work with express-jwt as per https://github.com/auth0/node-jwks-rsa/issues/308
Load initial `sub` from req.auth?.sub as per https://www.npmjs.com/package/express-jwt#migration-from-v6

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Check that you can go to the My Travel Requests page and create a travel request.
